cdk/domino_cdk/provisioners/eks/eks_iam_roles_for_k8s.py

```diff
@@ -0,0 +1,170 @@
+from typing import Dict, List
+
+import aws_cdk.aws_eks as eks
+import aws_cdk.aws_iam as iam
+from aws_cdk import core as cdk
+from aws_cdk.aws_s3 import Bucket
+
+# Permission groups
+
+s3_write_permissions = [
+    "s3:GetObject",
+    "s3:PutObject",
+    "s3:DeleteObject",
+    "s3:ListMultipartUploadParts",
+    "s3:AbortMultipartUpload",
+]
+
+s3_read_permissions = [
+    "s3:GetObject",
+]
+
+ecr_policies = {
+    "ecr_read_write_policy": [
+        "ecr:PutImageTagMutability",
+        "ecr:StartImageScan",
+        "ecr:ListTagsForResource",
+        "ecr:UploadLayerPart",
+        "ecr:BatchDeleteImage",
+        "ecr:ListImages",
+        "ecr:DeleteRepository",
+        "ecr:CompleteLayerUpload",
+        "ecr:TagResource",
+        "ecr:DescribeRepositories",
+        "ecr:DeleteRepositoryPolicy",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:ReplicateImage",
+        "ecr:GetLifecyclePolicy",
+        "ecr:PutLifecyclePolicy",
+        "ecr:DescribeImageScanFindings",
+        "ecr:GetLifecyclePolicyPreview",
+        "ecr:CreateRepository",
+        "ecr:PutImageScanningConfiguration",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:DeleteLifecyclePolicy",
+        "ecr:PutImage",
+        "ecr:UntagResource",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:StartLifecyclePolicyPreview",
+        "ecr:InitiateLayerUpload",
+        "ecr:GetRepositoryPolicy",
+    ]
+}
+
+# The bucket policies are auto-generated from the bucket list with the names write-name, read-name. E.g. write_blobs
+
+# Roles. The roles are the collection of the policies.
+roles = {
+    # E.g. nucleus needs this role
+    "write_blobs_read_logs": [
+        "write_blobs",
```

Comment:
The change from previous revision:
* role name is changed from "nucleus" to the name reflecting the contents of the role
* instead of crating one policy write_blobs_read_logs, we now create 2 separate policies read_blobs and write_blobs and add them to the role.